processor/src/main/java/com/blarley/rxretrobusseed/annotationprocessor/processor/RxRetroBusAnnotationProcessor.java

```diff
@@ -78,57 +78,74 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
             for (Element subElement : element.getEnclosedElements()) {
 
                 // ExecutableElements represent methods (among other things) - TODO: Figure out how this can break
-                if (subElement instanceof ExecutableElement && subElement.getAnnotation(Publish.class) != null) {
-
-                    // Cast to ExecutableElement in order to get Parameters
-                    ExecutableElement method = (ExecutableElement) subElement;
-                    String methodName = method.getSimpleName().toString();
-
-                    // Begin definition of method
-                    builder.append("\tpublic void " + methodName + "(");
-
-                    // Append parameters to method definition - TODO: Figure out how this can break
-                    String delim = "";
-                    StringBuilder params = new StringBuilder();
-                    StringBuilder args = new StringBuilder();
-                    for (VariableElement param : method.getParameters()) {
-                        params.append(delim)
-                                .append(param.asType() + " ")
-                                .append(param.getSimpleName().toString());
-
-                        args.append(delim)
-                                .append(param.getSimpleName().toString());
-                        delim = ", ";
+                if (subElement instanceof ExecutableElement) {
+
+                    FireAndForgetEvent fireAndForgetEvent = subElement.getAnnotation(FireAndForgetEvent.class);
+                    CachedEvent cachedEvent = subElement.getAnnotation(CachedEvent.class);
+                    UncachedEvent uncachedEvent = subElement.getAnnotation(UncachedEvent.class);
+
```

Comment:
Add a todo here - we need to validate that multiple event types are not on a method